napari/components/_viewer/model.py

```diff
@@ -265,7 +265,9 @@ def _new_markers(self):
         self.add_markers(empty_markers)
 
     def _new_shapes(self):
-        self.add_shapes([])
+        layer = self.add_shapes([])
+        layer.nd_data_dim = len(self.dims.indices)
+        layer.indices = self.dims.indices
```

Comment:
I feel like this is more fundamental than something you fiddle with after layer creation. I suggest something like `self.add_shapes([], dimensions=self.ndims)`. If I understand the dims model correctly, the idea is that *all* layers have access to it, either directly or through notifications, and this is how the shapes layer should know its dimensionality and current location.

Reply:
yes, this should be passed to the constructor. layers shouldn't really be changing their dimensions without actually changing their data